app/components/global_threads/thread_options/thread_option.tsx

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {
+    Text,
+    Platform,
+    TouchableHighlight,
+    TouchableNativeFeedback,
+    View,
+} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import {Theme} from '@mm-redux/types/theme';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+type Props = {
+    testID: string;
+    destructive: boolean;
+    icon: string;
+    onPress: () => void;
+    text: string;
+    theme: Theme;
+};
+
+function ThreadOption({destructive, icon, onPress, testID, text, theme}: Props) {
```

Comment:
is there a reason why not to use `post_option`? this seems to duplicate that component almost in its entirety, I'm fine if we want to migrate that component to TS though

Reply:
I get the reasoning but I wouldn't create a new component if there is no real need, in the future if things changed, we can look into adding a new component